File: src/http-client-mock/MultipartContentProvider.cs
using System.Linq;
using System.Net.Http;
using System.Net.Http.Json;
using JetBrains.Annotations;

namespace HttpClientMock
{
    public class MultipartContentProvider
    {
        private readonly MultipartContent content;

        public MultipartContentProvider(MultipartContent content)
        {
            this.content = content;
        }

        [PublicAPI]
        public T Get<T>(string name)
        {
            var x = content.First(c => c.Headers.ContentDisposition.Name == name);
            return x.ReadFromJsonAsync<T>().Result;
        }
    }
}

File: test/unit-test/HttpClientMockx/MultipartContentProvider.fs
namespace HttpClientMockx

open System.Net.Http
open System.Net.Http.Json
open JetBrains.Annotations

type MultipartContentProvider(content: MultipartContent) = 
    [<PublicAPI>]
    member this.Get<'T>(name: string): 'T =
        let x = content |> Seq.find (fun c -> c.Headers.ContentDisposition.Name = name)
        x.ReadFromJsonAsync<'T>().Result;
        
